empress/recon_vis/recon_viewer.py

```diff
@@ -96,83 +233,382 @@ def render_parasite_helper(fig, node, recon, host_lookup, show_internal_labels,
+    :param parasite_lookup: Dictionary with parasite node names as the key and parasite node objects as the values
+    """
+    node_pos = plot_tools.Position(node.layout.x, node.layout.y)
+
+    left_node, right_node = _get_children(node, recon_obj, parasite_lookup)
+
+    right_pos = plot_tools.Position(right_node.layout.x, right_node.layout.y)
+
+    mapping_node = recon_obj.mapping_of(node.name)
+    event = recon_obj.event_of(mapping_node)
+
+    if event.event_type is recon.EventType.COSPECIATION:
+        _render_cospeciation_branch(node, host_lookup, parasite_lookup, recon_obj, fig)
+    elif event.event_type is recon.EventType.DUPLICATION:
+        _connect_children(node, host_lookup, parasite_lookup, recon_obj, fig)
+    elif event.event_type is recon.EventType.TRANSFER:
+        _render_transfer_branch(node_pos, right_pos, fig, node, host_lookup, recon_obj, right_node)
+        _connect_child_to_parent(node, left_node, host_lookup, recon_obj, fig)
+    else:
+        raise ValueError('%s is not an known event type' % event.event_type)
+    # Losses are implicitly implied and are not mapped to any specific node
+
+
+def _connect_children(node: tree.Node, host_lookup: dict, parasite_lookup: dict, recon_obj: recon.Reconciliation, fig: plot_tools.FigureWrapper):
+    """
+    Connects the children of a node
+    :param node: Node object
+    :param host_lookup: Dictionary with host node names as the key and host node objects as the values
+    :param recon_obj: Reconciliation object
+    :param fig: Figure object that visualizes trees using MatplotLib
+    """
+    left_node, right_node = _get_children(node, recon_obj, parasite_lookup)
+    _connect_child_to_parent(node, left_node, host_lookup, recon_obj, fig)
+    _connect_child_to_parent(node, right_node, host_lookup, recon_obj, fig)
+
+
+def _render_loss_branch(node_pos: plot_tools.Position, next_pos: plot_tools.Position, fig: plot_tools.FigureWrapper):
+    """
+    Renders a loss branch given a two positions
+    :param node_pos: x and y position of a node
+    :param next_pos: x and y position of another node
+    :param fig: Figure object that visualizes trees using MatplotLib
+    """
+    # Create vertical line to next node
+    mid_pos = plot_tools.Position(node_pos.x, next_pos.y)
+    fig.line(node_pos, mid_pos, render_settings.LOSS_EDGE_COLOR, linestyle='--')
+    fig.line(mid_pos, next_pos, render_settings.PARASITE_EDGE_COLOR)
+
+
+def _render_cospeciation_branch(node: tree.Node, host_lookup: dict, parasite_lookup: dict, recon_obj: recon.Reconciliation, fig: plot_tools.FigureWrapper):
+    """
+    Renders the cospeciation branch.
+    :param node: Node object
+    :param host_lookup: Dictionary with host node names as the key and host node objects as the values
+    :param parasite_lookup: Dictionary with parasite node names as the key and parasite node objects as the values
+    :param recon_obj: Reconciliation object
```

Comment:
Either remove this line of comment or write a more detailed description.

Reply:
fixed